tests/test_x509.py

```diff
@@ -1420,6 +1421,49 @@ def test_build_cert(self, backend):
             x509.DNSName(u"cryptography.io"),
         ]
 
+    def test_build_cert_random_serial(self, backend):
+        sample_data = os.urandom(20)
+
+        @contextmanager
```

Comment:
`monkeypatch.reset()`

On Sat, Sep 3, 2016 at 10:49 AM, Paul Kehrer <notifications@github.com>
wrote:

> In tests/test_x509.py
> <https://github.com/pyca/cryptography/pull/3132#discussion_r77437689>:
>
> > @@ -1420,6 +1421,49 @@ def test_build_cert(self, backend):
> >              x509.DNSName(u"cryptography.io"),
> >          ]
> >
> > +    def test_build_cert_random_serial(self, backend):
> > +        sample_data = os.urandom(20)
> > +
> > +        @contextmanager
>
> Can't. That will patch from whenever you call it until end of the test,
> and since OpenSSL consumes randomness from os.urandom due to our random
> engine that means it will call the monkeypatched version when we don't want
> it to.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/pyca/cryptography/pull/3132/files/c4e01cc5e1c4b3bb402ce5bc35de2d88ffa597aa#r77437689>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAADBPFVzPJzkj0tDV2PXk8iU9XI5rNLks5qmYj_gaJpZM4J0Tse>
> .
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: D1B3 ADC0 E023 8CA6